Implement alternate console presentation

gitblit opened the pull request:
I'd like to propose an alternative to the current log-inspired console output.

Here is a comparison of `zzapi/examples/tests-bundle.zzb` with the current console output and this PR's alternative.

| Current | Proposal |
|--------|--------|
| <img width="1268" alt="image" src="https://github.com/agrostar/zzapi-cli/assets/868217/63ad64f2-537c-4c02-8281-ae18909d400d"> |  <img width="1096" alt="image" src="https://github.com/agrostar/zzapi-cli/assets/868217/01aaddb9-0cc9-42e6-88c4-80dbe229078c"> |

The objectives of this PR are to:
- introduce the concept of bundle name, currently taken from the filename
- use color to draw attention to the right pieces of information on a given line of console output

gitblit commented:
@vasan-agrostar @Varun0157 any thoughts on this?

Varun0157 commented:
It looks great to me - the reason we were using [info] and [fail] before was to provide syntax highlighting while using the `log` language for the vscode output panel, so what you've provided is a better format for the CLI. 

I feel #2 should be merged first because it deals with some pretty significant changes, and then we can deal with any conflicts while implementing your suggestions as required.

gitblit commented:
Sounds logical. I can rebase on your PR. 👍

I figured the log style was related to VS Code after I started reviewing that code and saw reuse of a log panel UI. I'm interested in that side too. Do you think an approach like presented in this PR is feasible? I don't know anything about VS Code plugin development.

vasan-agrostar commented:
@Varun0157 we need some changes in the output formatting when it comes to sub-objects (we are losing that information). I have reopened the related issue. Let us address that first, and then, @gitblit can rebase this PR. Also, if needed, change the presentation for tests as per convenience.

vasan-agrostar commented:
> Sounds logical. I can rebase on your PR. 👍
> 
> I figured the log style was related to VS Code after I started reviewing that code and saw reuse of a log panel UI. I'm interested in that side too. Do you think an approach like presented in this PR is feasible? I don't know anything about VS Code plugin development.

I'm afraid the approach in the VS Code output is not fully feasible. In VS Code, we can't control the colour as in the CLI. We have taken the approach of setting the output window's language to "log". This has the effect of syntax colouring the text based on keywords. FAIL and ERROR are coloured red and INFO is coloured green. The best we can do is play around within these constraints.

gitblit commented:
@vasan-agrostar @Varun0157 I've rebased and updated for the nested object tests.

| Current | Proposal |
|--------|-----------|
| <img width="668" alt="image" src="https://github.com/agrostar/zzapi-cli/assets/868217/fba6e4da-e50a-4045-b232-271509a818af"> | ![image](https://github.com/agrostar/zzapi-cli/assets/868217/1bb76f20-f6dd-4117-a37b-665623f6dbc8) |

vasan-agrostar commented:
@gitblit can you please confirm the rebase? I am not able to merge due to conflicts. Note that I have merged @Varun0157  's PRs:
1. Indented/flattened output for `$test` tests
2. Printing SKIP for `$skip` tests

gitblit commented:
> Printing SKIP for $skip tests

I'll adjust for this one over the weekend.

gitblit commented:
Rebased and adapted for `$skip` and `--indent`.

| Default | Indented |
|--------|----------|
| <img width="751" alt="image" src="https://github.com/agrostar/zzapi-cli/assets/868217/7677a912-6084-4f34-b9be-1e62b8e8bf85"> |  <img width="663" alt="image" src="https://github.com/agrostar/zzapi-cli/assets/868217/f427ce5f-ee5a-48e6-9afe-1e6fdcec1115"> |

gitblit commented:
Ready @vasan-agrostar @Varun0157

gitblit commented:
@vasan-agrostar I see the release failed and it appears it's because of Prettier. Sorry, I didn't notice anything about that. For some reason the checks didn't ever run on this PR. 🤔
